### PR TITLE
[Messenger] Improved exception on invalid message type

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

before:
invalid_message_type_before

after:
invalid_message_type_after

### CODE REVIEW

File: src/Symfony/Component/Messenger/MessageBus.php [hunk 1]
```diff
@@ -39,6 +39,10 @@ public function __construct(iterable $middlewareHandlers = array())
      */
     public function dispatch($message)
     {
+        if (!\is_object($message)) {
+            throw new \InvalidArgumentException(sprintf('Invalid type for message argument. Expected object, but got "%s".', \gettype($message)));
```
Comment (sroze) [inline src/Symfony/Component/Messenger/MessageBus.php]: Can you use an `InvalidArgumentException` coming from the Messenger package? (which will also implement Messenger's `ExceptionInterface`)

**Review by sroze — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Messenger/MessageBus.php (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Component/Messenger/MessageBus.php]: Done.

**Comment by yceruto** [comment]
Status: Needs Review

**Comment by ro0NL** [comment]
Looks like a safe guard to me =/

**Comment by sroze** [comment]
@ro0NL how would you improve the DX in this case then?

**Review by sroze — APPROVED** [review]

**Comment by ro0NL** [comment]
>  how would you improve the DX in this case then?

Depends.. how does `dispatch()` receive a string? If passed manually you are violating the contract already..

**Review by ogizanagi — APPROVED** [review]
Makes sense to me here and would save from weird exceptions in the profiler when the dispatched message wasn't an object, if it somehow reaches it.

**Review by Tobion — APPROVED** [review]

**Comment by sroze** [comment]
Thank you @yceruto.

**Pull request merged**